### ISSUE TITLE
Style Autoformatter 

### ISSUE THREAD

**Issue by heshamMassoud** [commercetools/commercetools-sync-java#293]
### Description

- It takes time for new developers to get acquainted with the new style rules. 
- It takes time to fix checkstyle errors. 
- It takes time to review style errors.
- There are a bunch of rules which can't be configured on checkstyle (e.g. chained method calls indentation, wrapping parameters indentation, etc..)
- check the comment here https://github.com/commercetools/commercetools-sync-java/pull/291#issuecomment-408360862

### Possible Solution
It would be great if there is an autoformatter that can fix checkstyle errors either on every commit, or on every grade build. 
Check: https://github.com/diffplug/spotless

**Comment by heshamMassoud** [comment]
One option is to use [Spotless](https://github.com/diffplug/spotless) gradle plugin with the [google-java-format](https://github.com/diffplug/spotless/tree/master/plugin-gradle#applying-to-java-source-with-google-java-format). However, it doesn't meet all the criteria 
✅  Independent of the IDE
❌  It would be nice if the style is configurable
❌  Style configuration should be version controlled. So the team can review any style configuration changes.
✅ Should be part of the build (before the compile step)

However, if we also agree on the style configuration decided by google. Then it would be ok for us, to have this style set there once and for all, and no more time wasted on style discussions or changes. 

I personally would be fine with that. What do you guys think? @butenkor @ahmetoz @ahalberkamp @lojzatran @LEQADA

**Comment by ahmetoz** [comment]
Google java style guide seems very popular on GitHub. In my opinion, we should use the [Google java style guide](http://google.github.io/styleguide/javaguide.html) as a standard and not create ours. And if we decide to use google java style guide we should refactor some code to make it consistent.

I have checked both ways like just using the Google way and manual customization. 
Spotless supports the eclipse settings but does not support the Intellij Idea's check style setting mechanism, there are issues about that. 

You can check with this code snippet to use google style guide with spotless.

``` java
spotless {
    java {
        googleJavaFormat()
        custom 'Lambda fix', { it.replace('} )', '})').replace('} ,', '},') }
        trimTrailingWhitespace()
        removeUnusedImports()
    }
    format 'misc', {
        target '**/.gitignore', '**/*.gradle', '**/*.md', '**/*.sh'
        indentWithSpaces(2)
        trimTrailingWhitespace()
        endWithNewline()
    }
}
```

**Comment by ahmetoz** [comment]
Resolved in #654
